src-plugins/mscAlgorithmPaint/mscAlgorithmPaintToolBox.cpp

```diff
@@ -615,39 +614,51 @@ void AlgorithmPaintToolBox::updateView()
     if (view)
     {
         setCurrentView(qobject_cast<medAbstractImageView*>(view));
-    }
-    updateMouseInteraction();
 
-    // TODO : get rid of similar lines in mousePressEvent, we need this here otherwise we cannot use interpolate/copy/paste
-    //                     //  on the view as long as the setData is not called for this view
-    if (currentView)
-    {
+        updateMouseInteraction();
+
         medAbstractData* data = currentView->layerData(0);
-        if(!data)
-            return;
-        medImageMaskAnnotationData * existingMaskAnnData = dynamic_cast<medImageMaskAnnotationData *>(data);
-        if(!existingMaskAnnData)
+
+        if(data)
         {
-            setData( data );
-        }
+            medImageMaskAnnotationData * existingMaskAnnData = dynamic_cast<medImageMaskAnnotationData *>(data);
+            if(!existingMaskAnnData)
+            {
+                setData( data );
+            }
 
-        // Update cursor if the orientation change
-        connect(currentView, SIGNAL(orientationChanged()), this, SLOT(activateCustomedCursor()), Qt::UniqueConnection);
+            // Update cursor if the orientation change
+            connect(currentView, SIGNAL(orientationChanged()), this, SLOT(activateCustomedCursor()), Qt::UniqueConnection);
 
-        connect(currentView,SIGNAL(layerRemoved(unsigned int)),this,SLOT(closeView()), Qt::UniqueConnection);
-    }
+            connect(currentView,SIGNAL(layerRemoved(unsigned int)),this,SLOT(clearToolBox()), Qt::UniqueConnection);
 
-    // Update cursor to new view
-    if ( this->m_strokeButton->isChecked() )
-    {
-        activateCustomedCursor(); // Add circular cursor for painting
+            // Update cursor to new view
+            if ( this->m_strokeButton->isChecked() )
+            {
+                activateCustomedCursor(); // Add circular cursor for painting
+            }
+            else
+            {
+                deactivateCustomedCursor(); // Deactivate painting cursor
+            }
+
+            disableButtons(false);
+        }
     }
     else
     {
-        deactivateCustomedCursor(); // Deactivate painting cursor
+        disableButtons(true);
+        currentView = NULL;
     }
 }
 
+void AlgorithmPaintToolBox::disableButtons(bool isToDisable)
```

Comment:
```isToDisable``` is a strange name :). Maybe ```disable``` is sufficient, and to be even clearer, you could take the same signature that QT uses (```setDisabled(bool disable)```). It's better because it makes it clear you are setting a state.